src/agent/rest_web_server.cpp

```diff
@@ -0,0 +1,1061 @@
+    cJSON_AddItemToObject(ret, "state",cState);
+    cJSON_AddItemToObject(ret, "extAddress",cExtAddr );
+    cJSON_AddItemToObject(ret, "rloc16",cRloc16 );
+    cJSON_AddItemToObject(ret, "numOfRouter",cNumRoute );
+    cJSON_AddItemToObject(ret, "leaderData",cLeaderData );
+    cJSON_AddItemToObject(ret, "extPanId",cExtPanId );
+
+    return ret;
+}
+
+cJSON*  RestWebServer::getJsonExtendedAddr(Connection* connection, otInstance * mInstance){
+    connection->hasCallback = 0;
+    cJSON* cExtAddr;
+
+    const uint8_t *extAddress = reinterpret_cast<const uint8_t *>(otLinkGetExtendedAddress(mInstance));
+    char* p = formatBytes(extAddress, OT_EXT_ADDRESS_SIZE);
+    cExtAddr  = cJSON_CreateString(p);
+
+    return cExtAddr;
+    
+ }
+
+cJSON* RestWebServer::getJsonState(Connection* connection,otInstance * mInstance){
+    connection->hasCallback = 0;
+    cJSON * cState;
+    
+
+    switch (otThreadGetDeviceRole(mInstance))
+        {
+        case OT_DEVICE_ROLE_DISABLED:
+            cState = cJSON_CreateString("disabled");
+            break;
+        case OT_DEVICE_ROLE_DETACHED:
+            cState = cJSON_CreateString("detached");
+            break;
+        case OT_DEVICE_ROLE_CHILD:
+            cState = cJSON_CreateString("child");
+            break;
+        case OT_DEVICE_ROLE_ROUTER:
+            cState = cJSON_CreateString("router");
+            break;
+        case OT_DEVICE_ROLE_LEADER:
+            cState = cJSON_CreateString("leader");
+            break;
+        default:
+            cState = cJSON_CreateString("invalid state");
+            break;
+        }
+    
+    
+    return cState;
+}
+
+cJSON* RestWebServer::getJsonNetworkName(Connection* connection,otInstance * mInstance){
+    connection->hasCallback = 0;
+    const char *networkName = otThreadGetNetworkName(mInstance);
+    char *p = new char[20]();
+    sprintf(p,"%s",static_cast<const char *>(networkName));
+    cJSON* ret = cJSON_CreateString(p);
+    
+    return ret;
+}
+
+cJSON*  RestWebServer::getJsonLeaderData(Connection* connection,otInstance * mInstance){
+    connection->hasCallback = 0;
+    otLeaderData leaderData;
+    otThreadGetLeaderData(mInstance, &leaderData);
+    cJSON* ret = CreateJsonLeaderData(leaderData);
+    return ret;
+
+}
+
+cJSON*  RestWebServer::getJsonNumOfRoute( Connection* connection, otInstance * mInstance){
+    connection->hasCallback = 0;
+    int count = 0;
+    uint8_t maxRouterId;
+    otRouterInfo routerInfo;
+    maxRouterId = otThreadGetMaxRouterId(mInstance);
+    char *p = new char[6]();
+     
+    for (uint8_t i = 0; i <= maxRouterId; i++){
+            if (otThreadGetRouterInfo(mInstance, i, &routerInfo) != OT_ERROR_NONE)
+            {
+                continue;
+            }
+            count++;
+    }
+    sprintf(p,"%d",count);
+    cJSON * ret = cJSON_CreateString(p);
+    
+    return ret;
+
+}
+cJSON*  RestWebServer::getJsonRloc16(Connection* connection, otInstance * mInstance){
```

Comment:
Do we need the argument `connection`?

Reply:
Yes, need to set some parameters about connection